gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALException.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm.gdal
+
+
+/** Base class for all Exceptions involving GDAL. */
+class GDALException(message: String) extends Exception(message)
```

Comment:
Should it be a `RuntimeException` like in `geotrellis-gdal`? I don't have a strong opinion about that btw.

Reply:
@pomadchin `RuntimeException` should only be used for _unchecked_ exceptions (in a Java sense). While all exceptions in Scala are unchecked, I'd suggest following the general intent, and keeping `Exception` as the base type.

> RuntimeException is the superclass of those exceptions that can be thrown during the normal operation of the Java Virtual Machine.
> RuntimeException and its subclasses are unchecked exceptions. Unchecked exceptions do not need to be declared in a method or constructor's throws clause if they can be thrown by the execution of the method or constructor and propagate outside the method or constructor boundary.

Reply:
That said.... I could see debates on both side of the issue, so I wouldn't fight over my position ;-)

Reply:
@metasim yea in Scala all exceptions are unchecked, but I was just curious does it make too much sense to follow Java logic or not (like MalformedArgument exceptions, etc). But I see your comment about IO and I'm good with keeping it `<: Exception`